app/src/main/kotlin/com/wire/android/ui/authentication/login/sso/LoginSSOViewModel.kt

```diff
@@ -44,6 +60,49 @@ class LoginSSOViewModel @Inject constructor(
         }
     }
 
+    @VisibleForTesting
+    fun establishSSOSession(ssoLoginResult: DeepLinkResult.SSOLogin.Success) {
+        viewModelScope.launch {
+            //TODO: serverConfig should be fetched of the serverConfigRepository
+            val authSession = getSSOLoginSessionUseCase(ssoLoginResult.cookie, ServerConfig.STAGING)
+                .let {
+                    when (it) {
+                        is SSOLoginSessionResult.Failure -> {
+                            return@launch
+                        }
+                        is SSOLoginSessionResult.Success -> it.userSession
+                    }
+                }
+            val storedUserId = addAuthenticatedUser(authSession, false).let {
+                when (it) {
+                    is AddAuthenticatedUserUseCase.Result.Failure -> {
+                        return@launch
+                    }
+                    is AddAuthenticatedUserUseCase.Result.Success -> it.userId
+                }
+            }
+            registerClient(storedUserId).let {
+                when (it) {
+                    is RegisterClientResult.Failure -> {
+                        return@launch
+                    }
+                    is RegisterClientResult.Success -> navigateToConvScreen()
+                }
+            }
```

Comment:
We still need to cover all use cases by unit test.
what if `getSSOLoginSessionUseCase` and `addAuthenticatedUser` return failure, how the app should act ?

Reply:
Correct, I'm adding them as well.